fix(datepicker): fixed the datepicker component bidi layout issue. (#8161)

DavidSCChen opened the pull request:
Closes #8161 

Fixed the next & previous month arrow orientation for datepicker component in bidi(RTL) layout by adding the RTLCSS value directives.

#### Changelog

**New**
Added RTLCSS value directives in to the _flatpickr.scss.

#### Testing / Reviewing

Verified the fix has no impact to the ltr layout by using the local storybook development server.
Also created the sandbox to verify the rtl layout:
https://codesandbox.io/s/webpack-react-rtlcss-carbon-datepicker-fixed-hpe4s?file=/app/carbon-components.scss:239237-239282

DavidSCChen committed:
fix(datepicker): fixed the datepicker bidi layout issue. (#8161)

Fixed the next & previous month arrow orientation for datepicker component in bidi(RTL) layout by adding the RTLCSS value directives. (#8161)

tw15egan commented on packages/components/src/components/date-picker/_flatpickr.scss:
```diff
@@ -142,6 +142,7 @@
   width: rem(40px);
   padding: 0;
   fill: $icon-01;
+  transform: scale(1, 1) /* rtl: scale(-1,1) */;
```
Due to a limitation with the way sass compiles RTL directives, we need to change the format to something like this so it is correctly applied 

```suggestion
  transform: scale(1, 1) #{'/*rtl: scale(-1,1)*/'};
```

This way, sass correctly interpolates the comment as a directive 

See usage in Checkbox: https://github.com/carbon-design-system/carbon/blob/main/packages/components/src/components/checkbox/_checkbox.scss#L105

DavidSCChen replied:
Hi @tw15egan ,
Thanks for your review, I had updated the code as suggested.

DavidSCChen committed:
fix(datepicker): fixed the datepicker bidi layout issue. (#8161)

Wrapped the RTL directive with  #{}, so sass could correctly interpolates the comment as a directive (#8161)

kodiakhq[bot] committed:
Merge branch 'main' into fix-8161-datepicker-component-rtlcss
